pkg/controllers/nodeclaim/termination/controller.go

```diff
@@ -89,14 +91,41 @@ func (c *Controller) Finalize(ctx context.Context, nodeClaim *v1beta1.NodeClaim)
 		return reconcile.Result{}, nil
 	}
 	if nodeClaim.Status.ProviderID != "" {
-		if err = c.cloudProvider.Delete(ctx, nodeClaim); cloudprovider.IgnoreNodeClaimNotFoundError(err) != nil {
-			// We expect cloudProvider to emit a Retryable Error when the underlying instance is not terminated and if that
-			// happens, we want to re-enqueue reconciliation until we terminate the underlying instance before removing
-			// finalizer from the nodeClaim.
-			if cloudprovider.IsRetryableError(err) {
-				return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+		// Check if the status condition on nodeClaim is Terminating
+		if nodeclaimutil.GetCondition(nodeClaim, v1beta1.Terminating).Status != v1.ConditionTrue {
+			// If not then call Delete on cloudProvider to trigger termination
+			if err = c.cloudProvider.Delete(ctx, nodeClaim); cloudprovider.IgnoreNodeClaimNotFoundError(err) != nil {
+				return reconcile.Result{}, fmt.Errorf("terminating cloudprovider instance, %w", err)
+			}
+			// Set the status condition on nodeClaim as Terminating
+			nodeClaim.StatusConditions().SetCondition(apis.Condition{
+				Type:     v1beta1.Terminating,
+				Status:   v1.ConditionTrue,
+				Severity: apis.ConditionSeverityInfo,
+			})
+			if !equality.Semantic.DeepEqual(stored, nodeClaim) {
```

Comment:
You don't need to do an equality check if you are doing an Update operation, so long as the operation itself is gated. Since you are gating the addition of the status condition here based on whether the condition is already set, I don't actually think you need this